src/3id/index.js

```diff
@@ -54,6 +64,54 @@ class ThreeId {
     })
   }
 
+  async _initDID (muportIpfs) {
+    const muportPromise = this._initMuport(muportIpfs)
+    const { did, cid } = await this._init3ID()
+    this._rootDID = did
+    const spaces = Object.keys(this._keyrings)
+    const subDIDs = await Promise.all(
+      spaces.map(space => {
+        return this._init3ID(space)
+      })
+    )
+    this._subDIDs = {}
+    spaces.map((space, i) => {
+      this._subDIDs[space] = subDIDs[i].did
+    })
+    await muportPromise
+  }
+
+  async _init3ID (spaceName) {
+    const doc = new DidDocument(this._ipfs, DID_METHOD_NAME)
+    if (!spaceName) {
+      const pubkeys = this._mainKeyring.getPublicKeys()
+      doc.addPublicKey('signingKey', 'Secp256k1VerificationKey2018', 'publicKeyHex', pubkeys.signingKey)
+      doc.addPublicKey('encryptionKey', 'Curve25519EncryptionPublicKey', 'publicKeyBase64', pubkeys.asymEncryptionKey)
+      doc.addPublicKey('managementKey', 'Secp256k1VerificationKey2018', 'ethereumAddress', this.managementAddress)
+      doc.addAuthentication('Secp256k1SignatureAuthentication2018', 'signingKey')
+    } else {
+      const pubkeys = this._keyrings[spaceName].getPublicKeys()
```

Comment:
when this is called, there will always be a keyring available by then?

Reply:
It should be yes.